include/RAJA/policy/cuda/policy.hpp

```diff
@@ -1248,16 +1408,147 @@ using policy::cuda::unordered_cuda_loop_y_block_iter_x_threadblock_average;
 using policy::cuda::cuda_atomic;
 using policy::cuda::cuda_atomic_explicit;
 
+
 // policies usable with reducers
-using policy::cuda::cuda_reduce_device_fence;
-using policy::cuda::cuda_reduce_block_fence;
-using policy::cuda::cuda_reduce_atomic_device_init_device_fence;
-using policy::cuda::cuda_reduce_atomic_device_init_block_fence;
-using policy::cuda::cuda_reduce_atomic_host_init_device_fence;
-using policy::cuda::cuda_reduce_atomic_host_init_block_fence;
-using policy::cuda::cuda_reduce_base;
-using policy::cuda::cuda_reduce;
-using policy::cuda::cuda_reduce_atomic;
+template < cuda::reduce_algorithm algorithm,
+           cuda::block_communication_mode comm_mode,
+           size_t replication = named_usage::unspecified,
+           size_t atomic_stride = named_usage::unspecified >
+using cuda_reduce_tuning = policy::cuda::cuda_reduce_policy<
+    cuda::ReduceTuning<algorithm, comm_mode, replication, atomic_stride>>;
+
+// Policies for RAJA::Reduce* objects with specific behaviors.
+// - non-atomic policies store partial results and combine them in the same
+//   order every time, leading to consistent results for a loop run to run.
+// - *atomic* policies may use atomics to combine partial results. The
+//   use of atomics leads to order of operation differences which change the
+//   results of floating point sum reductions for a loop run to run. Falls back
+//   on a non-atomic implementation if atomics can't be used with the given
+//   type. The memory used with atomics is initialized on the device using
+//   atomics which adds overhead.
+// - *atomic_host* policies are similar to the atomic policies above. However
+//   the memory used with atomics is initialized on the host. This is faster
+//   overall than other policies on HW with direct host access to device memory
+//   such as the IBM power 9 + Nvidia V100 Sierra/Lassen systems.
+// - *device_fence policies use normal memory accesses with device scope fences
+//                in the implementation. This works on all HW.
+// - *block_fence policies use special (atomic) memory accesses that only cache
+//                 in a cache shared by the whole device to avoid having to use
+//                 device scope fences. This improves performance on some HW but
+//                 is more difficult to code correctly.
```

Comment:
```suggestion
// - *device_fence* policies use normal memory accesses with device scope fences
//                in the implementation. This works on all HW.
// - *block_fence* policies use special (atomic) memory accesses that use
//                 a cache shared by the whole device to avoid having to use
//                 device scope fences. This improves performance on some HW but
//                 is more difficult to code correctly.
```